### PR TITLE
Add opentelemetry as tracing provider

### LINKED ISSUE #1005: Add opentelemetry as tracing provider
See the [trace package](https://github.com/virtual-kubelet/virtual-kubelet/tree/master/trace) which currently defines an interface as well as an opencensus implementation.

### PR DESCRIPTION
**Pull request by yabuchan**
**What type of PR is this?**
/feature

**What this PR does / why we need it:**
Fixes #1005

**Special notes for your reviewer:**
* Currently opentelemetry hasn't supported log signal in span, so left TODO comment as saving log in span.
* Latest opentelemetry (v1.7.0) requires go-logr/logr v1.2 which conflicts with k8s.io/apimachinery. Safe to bump up to opentelemetry (v1.7.0) once k8s.io/apimachinery is updated. 

**How was it tested:**
`make test` is passed

### CODE REVIEW

File: trace/opentelemetry/opentelemetry.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+// Package openTelemetry implements a github.com/virtual-kubelet/virtual-kubelet/trace.Tracer
+// using openTelemetry as a backend.
+//
+// Use this by setting `trace.T = Adapter{}`
+//
+// For customizing trace provider used in Adapter, set trace provider by
+//`otel.SetTracerProvider(*sdktrace.TracerProvider)`. Examples of customize are setting service name,
+// use your own exporter (e.g. jaeger, otlp, prometheus, zipkin, and stdout) etc. Do not forget
+// to call TracerProvider.Shutdown() when you create your TracerProvider to avoid memory leak.
+package openTelemetry
```
Comment (cpuguy83) [inline trace/opentelemetry/opentelemetry.go:24]:
```suggestion
package opentelemetry
```

File: trace/opentelemetry/opentelemetry.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+		l.l.Errorf(f, args...)
+	case lFatal:
+		l.l.Fatalf(f, args...)
+	}
+
+	if !l.s.IsRecording() {
+		return
+	}
+
+	//TODO Add log event to span once it is available.
```
Comment (cpuguy83) [inline trace/opentelemetry/opentelemetry.go:211]: This would be `span.AddEvent`

Comment (cpuguy83) [reply trace/opentelemetry/opentelemetry.go:211]: Or at least that is how I've done this in other projects.

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: trace/opentelemetry/opentelemetry.go (hunk 1 above)
Comment (yabuchan) [reply trace/opentelemetry/opentelemetry.go]: good catch. Updated.

**Comment by cpuguy83** [comment]
```
trace/opentelemetry/opentelemetry.go:15:1: commentFormatting: put a space between `//` and comment text (gocritic)
// Package openTelemetry implements a github.com/virtual-kubelet/virtual-kubelet/trace.Tracer
^
trace/opentelemetry/opentelemetry_test.go:45:3: ineffectual assignment to `ctx` (ineffassign)
		ctx, s := a.StartSpan(ctx, "name")
		^
trace/opentelemetry/opentelemetry_test.go:185:4: ineffectual assignment to `ctx` (ineffassign)
			ctx = s.WithFields(ctx, tt.fields)
			^
trace/opentelemetry/opentelemetry_test.go:257:4: ineffectual assignment to `ctx` (ineffassign)
			ctx, s := otel.Tracer(tt.spanName).Start(ctx, tt.spanName)
			^
trace/opentelemetry/opentelemetry.go:51:19: func `logLevel.string` is unused (unused)
```

**Comment by cpuguy83** [comment]
This is great.
I've been holding off merging due to CI issues.
I have a pr open to address that.

Could you have the cli use this instead of opencensus? I think we can just add in otlp support in the cli and drop all the opencensus stuff.

**Pull request merged**